### PR TITLE
Time synchronization between worlds

### LINKED ISSUE #42: Option for synchronized time
Would define a time offset in each region. If present, a recurring task would set all regions to a synced "global time" + that offset. This would allow us to keep a proper day-night cycle between regions.

### PR DESCRIPTION
**Pull request by Norazan**
Closes #42

### CODE REVIEW

File: src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt [hunk 1]
```diff
@@ -33,5 +35,31 @@ object DeeperCommandExecutor : IdofrontCommandExecutor() {
                     sender.info("${player.name} is in section ${section.key}")
             }
         }
+        "syncedtime" {
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt:38]: Maybe we should start putting these commands under one "deeperworld" command? I think sectionon and secitonoff should probably be under it as well, maybe even do a `/deeperworld tp on/off` with an arg.

File: src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt [hunk 2]
```diff
@@ -33,5 +35,31 @@ object DeeperCommandExecutor : IdofrontCommandExecutor() {
...
+            "set"{
+                val time by intArg()
+                action {
+                    DeeperConfig.data.time.mainWorld?.let {
+                        it.time = time.toLong()
+                    } ?: return@action
+
+                    DeeperConfig.data.time.syncedWorlds.forEach {
+                        it.key.time = time.toLong() + it.value
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt:47]: Just btw, you can do `forEach { (key, value) -> ... }` to name the key and value so they're a little more clear. Maybe use `(world, offset)` here.

File: src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt [hunk 3]
```diff
@@ -33,5 +35,31 @@ object DeeperCommandExecutor : IdofrontCommandExecutor() {
...
+                    }
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt:48]: If you want this to be saved to the config, you should be able to do `DeeperConfig.queueSave()` or something similar, but that just reserializes the entire structure and I forget if it preserves comments or not... maybe I should figure out something for that 🤔.

You can ignore this for now but I guess just know this won't be persisted between restarts.

File: src/main/java/com/derongan/minecraft/deeperworld/config/TimeConfig.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import com.mineinabyss.idofront.serialization.WorldSerializer
+import com.mineinabyss.idofront.time.TimeSpan
+import com.mineinabyss.idofront.time.TimeSpanSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.UseSerializers
+import org.bukkit.World
+
+@Serializable
+class TimeConfig(
+    val updateInterval: @Serializable(with = TimeSpanSerializer::class) TimeSpan = TimeSpan(
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/config/TimeConfig.kt:14]: Should be able to do 1800.seconds for the TimeSpan object. Sometimes intellij forces you to do (1800).seconds for the autocomplete to show it, but you can remove the brackets after you do the import.

**Review by 0ffz — CHANGES_REQUESTED** [review]
Small notes for the commands but looks good otherwise

File: src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt (hunk 1 above)
Comment (Norazan) [reply src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt:38]: Maybe make it a shorthand syntax instead? `/dw time set 1000` for example. Bit easier to use.

Comment (0ffz) [reply src/main/java/com/derongan/minecraft/deeperworld/DeeperCommandExecutor.kt:38]: Yeah you can have both with `("deeperworld" / "dw") { ... }`

**Review by 0ffz — APPROVED** [review]

**Pull request merged**